lib/Conversion/LLHDToLLVM/LLHDToLLVM.cpp

```diff
@@ -165,8 +164,14 @@ static LLVM::LLVMType getProcPersistenceTy(LLVM::LLVMDialect *dialect,
   SmallVector<LLVM::LLVMType, 3> types = SmallVector<LLVM::LLVMType, 3>();
   proc.walk([&](Operation *op) -> void {
     if (op->isUsedOutsideOfBlock(op->getBlock())) {
-      types.push_back(converter.convertType(op->getResult(0).getType())
-                          .cast<LLVM::LLVMType>());
+      if (auto ptr = op->getResult(0).getType().dyn_cast<PtrType>()) {
+        // Persist the unwrapped value.
+        auto converted = converter.convertType(ptr.getUnderlyingType());
+        types.push_back(converted.cast<LLVM::LLVMType>());
+      } else {
+        types.push_back(converter.convertType(op->getResult(0).getType())
+                            .cast<LLVM::LLVMType>());
+      }
```

Comment:
So essentially we want to migrate stack slots into the process' persistent memory *by value*, such that they are retained across `wait` suspension points -- pretty much like yield/resume coroutines, right? Since we're only deciding based on whether the instruction result is a pointer, this also allocates memory for things like `extf`/`exts` operating on pointers?

Reply:
Yes exactly, this pretty much mimics coroutines behavior I think. They would probably be a better option long-term.
Also, that's a good point, this does not make a distinction for `extf`/`exts` operating on pointers, so a new slot will be created in the local state, that differs from the original pointer!

Reply:
So that means if I do the following:
```
^bb0:
    %0 = const i32 0
    %1 = var i32 %0
    %2 = exts i8* from i32* %1, 0, 8
    wait ^bb1
^bb1:
    %3 = const i8 42
    store i8* %2, %3
```
The store would be absorbed in the persistent memory for `%2`, but not in the lower 8 bits of `%1`, right?

Maybe a fix could be to decidedly only allocate persistent memory for `var` instructions?

Reply:
> The store would be absorbed in the persistent memory for `%2`, but not in the lower 8 bits of `%1`, right?

Yes, exactly.

> Maybe a fix could be to decidedly only allocate persistent memory for `var` instructions?

Kind of I think, we still need to restore `%2` in some manner, maybe by storing the shifted pointer into the local state directly, instead of the value?

Reply:
Yeah I think that would be a good option. Because we essentially promote the `var` stack allocation to a "heap" allocation in the persisted memory, the shifted pointer should remain valid.

Reply:
So, turns out adding pointer support to the `extf`/`exts` operations completely flew under my radar, so we cannot have pointers generated from such operations at all currently. I think we'll be better off solving this in a future patch after actually adding support for pointers to the operations, maybe together with tweaking the lowered pointer representation to fully represent the bit-precision of those operations?

Reply:
Yeah I agree, that solves the issue quite nicely for now :stuck_out_tongue:. If you have the `extf`/`exts` for signals already, expanding those to pointers will be an easy future extension.

Reply:
Yes, using pointer-offset structs similar to signals should make it easy to reuse that part of the implementation!